Move away from travis

ponceta opened the pull request:
On basis work of @m-kuhn,

Need cleaning of scripts/ci folder which is now useless

ponceta commented:
Need to fix gettext and some Makefile functions to make it work smoothly. I'll try to get into sphinx and make these work on my computer before further changes.

ponceta commented on .github/workflows/release.yml:
```diff
@@ -0,0 +1,35 @@
+          sudo apt-get install -y gettext
+
+      - name: tx push / pull
+        if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/move_away_from_travis' }}
```
ping  @m-kuhn @ponceta

ponceta commented:
![image](https://user-images.githubusercontent.com/8614223/152293185-7b35c4d0-2eff-4129-b5c6-c51d483ce111.png)

![image](https://user-images.githubusercontent.com/8614223/152292283-8f2b78bf-7d7a-4541-97c9-ad5649145149.png)

Transifex has been pushed and pulled correctly. We still need to fix the syntax :

Existing : 
`PO-admin-guide--data_manager`
Newly created
`PO-i18n_pot_admin-guide_data_manager`

@m-kuhn it seems to be mandatory in the TX config file (even if it's a burden for maintenance)?

Or can we rename ressources in transifex to match a certain pattern?

m-kuhn commented:
These files have been created by a test from me yesterday. I think we should be able to avoid listing the resources.

ponceta commented:
@m-kuhn I've fixed on transifex side the ressources names (and removed the i18n files) so that they match with what is pushed from GitHub.

I see that translated chains or translated pages are not either pulled or build correctly what results in untranslated pages like : 
https://qwat.github.io/docs/fr/admin-guide/data_manager.html

ponceta commented:
make transifex_pull seems to have difficulties on qwat_docs

![image](https://user-images.githubusercontent.com/8614223/152509437-48ddb6e1-6fcc-49e8-a3f8-b45f53b06d3e.png)

m-kuhn commented:
@ponceta are you taking this over or should I still have a look?

ponceta commented:
@m-kuhn I could use a hand, I feel I miss something since I can retrieve, translate and build qgep_doc locally but not qwat_doc.

ponceta commented:
![image](https://user-images.githubusercontent.com/8614223/152520712-dc825346-f621-4223-a6e6-71817e6d3d0c.png)
index.html has not been updated since three days, maybe I miss something with the deploy process :S

ponceta commented on .github/workflows/release.yml:
```diff
@@ -0,0 +1,34 @@
+          touch build/html/.nojekyll
+
+      - name: Deploy 🚀
+        if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/move_away_from_travis' }}
```
Same remark as before

m-kuhn commented:
@ponceta I will have a look end of this morning, ok for you?

ponceta commented:
@m-kuhn this would be great! I really hope we can make this work! We will be then able to  focus on content and translations! :)

m-kuhn commented:
Sorry I was blocked in the morning. Later today.

m-kuhn commented:
> index.html has not been updated since three days, maybe I miss something with the deploy process :S

It looks like other files in the same place have been updated properly. Very likely this file simply did not change (so no reason to update).

m-kuhn commented on .tx/config:
```diff
@@ -1,3 +1,245 @@
 host = https://www.transifex.com
 
+[qwat-doc.admin-guide-data-manager]
+source_file = i18n/pot/admin-guide/data-manager.pot
```
@ponceta this does not match the actual filename (`data_manager.pot`), was this done on purpose?

m-kuhn replied:
FYI, I have a commit sitting here ready to completely generate this file, but only if you can confirm this should be `data_manager.pot`.

ponceta replied:
This is indeed not the right syntax if we want to map data_manager.rst file. I let you check this and the other files (.pot and po) below if you have a commit ready.

data-manager -> data_manager

m-kuhn commented on scripts/build-docs.sh:
```diff
@@ -1,45 +1,14 @@
-#!/usr/bin/env bash
+#!/bin/bash
```
That was actually better the other way (more portable)

ponceta replied:
```suggestion
#!/usr/bin/env bash
```
Let's revert (copy from QGEP -> adapt on QGEP ;)

m-kuhn commented on README.md:
```diff
@@ -1,24 +1,20 @@
-CI](https://travis-ci.org/qwat/docs)) and published on the links above.
+The latest version of the documentation is automatically built (using Github Actions) and published on the links above.
 
 Translation is done via transifex.
```
Link to the qwat transifex project?
Keeping the entry barrier for contributors as small as possible.
```suggestion
Translation is done via transifex, [help translating by joining the team](https://www.transifex.com/qwat/qwat-doc/).
```

ponceta commented on .tx/config:
```diff
@@ -1,3 +1,245 @@
+type = PO
+
+[qwat-doc.developer-guide-index]
+source_file = i18n/pot/developer/guide-index.pot
```
```suggestion
source_file = i18n/pot/developer-guide/index.pot
```

ponceta commented on .tx/config:
```diff
@@ -1,3 +1,245 @@
+type = PO
+
+[qwat-doc.developer-guide-local-customizations]
+source_file = i18n/pot/developer-guide/local-customizations.pot
```
```suggestion
source_file = i18n/pot/developer-guide/local_customizations.pot
```

ponceta commented on .tx/config:
```diff
@@ -1,3 +1,245 @@
+type = PO
+
+[qwat-doc.developer-guide-qwat-dev]
+source_file = i18n/pot/developer-guide/qwat-dev.pot
```
```suggestion
source_file = i18n/pot/developer-guide/qwat_dev.pot
```

ponceta commented on .tx/config:
```diff
@@ -1,3 +1,245 @@
+type = PO
+
+[qwat-doc.features-guide-index]
+source_file = i18n/pot/features/guide-index.pot
```
```suggestion
source_file = i18n/pot/features-guide/index.pot
```

ponceta commented on .tx/config:
```diff
@@ -1,3 +1,245 @@
+type = PO
+
+[qwat-doc.installation-guide-database-initialization]
+source_file = i18n/pot/installation-guide/database-initialization.pot
```
This is actually OK, but we could change the rst file name with an underscore if we want to be systematic with files namings

m-kuhn replied:
That is a good idea, I'm just not sure if transifex will transfer all the existing translations or if that needs manual work.

ponceta commented:
@m-kuhn this seems fine in the build and actions process but actually still does not the job of applying the translations to the files :
https://qwat.github.io/docs/fr/features-guide/index.html as an example.

m-kuhn commented:
yep, I am looking at this

ponceta commented:
i18n_pot_* ressources have been pushed again toward transifex, maybe that's where the translation is stuck :S
ex :
i18n_pot_admin-guide_data_manager

m-kuhn commented:
@ponceta this looks good to me now:

https://qwat.github.io/docs/fr/user-guide/update.html

m-kuhn commented:
The current approach is to store the complete .tx/config file in the source. It will be automatically generated through github actions, so we should always have an up to date version. This is not really a perfect workflow but should be good enough for the moment.

Also note that we have other hard to discover parts of the system. E.g. on transifex there is a high percentage of Czech (cs) translation, however each language needs to be added to the Makefile. Cs is added now.

ponceta commented:
Aie, @m-kuhn I should have warned you, QWAT doc is partly still not available in english...

It is mostly the case for the user guide:
https://qwat.github.io/docs/fr/user-guide/update.html

But if you look at a translated en->fr page the translation workflow does not work
https://qwat.github.io/docs/fr/features-guide/index.html

Same for Czech :

https://qwat.github.io/docs/cs/features-guide/index.html

(although translations are available on transifex and correctly downloaded if you try it locally)

None of the translated html pages has been pushed to gh-page : (only doc-tree refresh)

![image](https://user-images.githubusercontent.com/8614223/153551800-a9abce95-6be9-41bb-900c-a8bea9b27e28.png)

m-kuhn commented:
Thanks for the feedback and sorry for missing that.
I can confirm translated files (.po) get pulled from transifex, get compiled (.mo) but are not integrated in the sphinx-build.

ponceta commented:
Is it possible that the fact that we have several index.po files could make trouble?

![image](https://user-images.githubusercontent.com/8614223/153605358-77f62845-d4a0-461d-a587-e7a321bd805c.png)

I was not able to check the *.mo files content to see if all translations are well compiled.

m-kuhn commented:
Confirmed, this is exactly the problem.
We need to setup the equal directory structure

m-kuhn commented on README.md:
```diff
@@ -1,24 +1,20 @@
+ * [Deutsch](https://qwat.github.io/docs/de/)
+ * [Français](https://qwat.github.io/docs/fr/)
+ * [Italiano](https://qwat.github.io/docs/it/)
+ * [Română](https://qwat.github.io/docs/ro/)
```
```suggestion
 * [Română](https://qwat.github.io/docs/ro/)
 * [Czech](https://qwat.github.io/docs/cs/)
```

m-kuhn commented on scripts/create_tx_conf.py:
```diff
@@ -0,0 +1,25 @@
+# -*- coding: utf-8 -*-
+
+# Creates a transifex configuration from pot files that are found in i18n/pot
```
```suggestion
# Creates a transifex configuration from pot files that are found in i18n/pot
# Will be automatically run and committed via github actions
```

ponceta commented on .github/workflows/release.yml:
```diff
@@ -0,0 +1,38 @@
+          touch build/html/.nojekyll
+
+      - name: Deploy 🚀
+        if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/maste' }}
```
```suggestion
        if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/master' }}
```
Typo : master

ponceta commented:
Thank you @m-kuhn ! 👍
